### DISCUSSION TITLE
How does uBlock Origin resolve having multiple scriptlets with the same name?

### DISCUSSION THREAD

**Discussion by partingscientist** [uBlockOrigin/uBlock-issues#2571] [Q&A] [answered]
I thought I'd rather ask instead of scouring the code base or testing myself and getting partial answers.

I'd assume that
1. the built-in scriptlets are loaded first before `userResourcesLocation`'s scriptlets and any duplicate scriptlets loaded from `userResourcesLocation` will override the built-in ones; 
2. resources from `userResourcesLocation` are loaded in order of entry and any duplicate scriptlets will override those imported from resources declared earlier. 

Are both assumptions correct? In addition, did I miss anything that may be useful? 

Thank you.

**Comment by ghost** [comment, marked as answer]
https://github.com/uBlockOrigin/uBlock-issues/wiki/Advanced-settings#userresourceslocation

**Reply by partingscientist** [reply]
Somehow missed this! Thank you.
